R/function_argument_linter.R

```diff
@@ -0,0 +1,29 @@
+#' Function argument linter
+#'
+#' Check that arguments with defaults come last in all function declarations.
```

Comment:
Ditto -- external link to the design guide

Reply:
done.